check doesn't respect runxfail option

When `runxfail` option is set (`pytest --runxfail ...`) tests that use `check` are reported as xfail instead of failed:

_xfail_test.py_
```python
import pytest

@pytest.mark.xfail(reason="foo")
def test_xfail(check):
    with check:  # with this line gone, this is actual behaviour
        assert False
``` 

# Expected behaviour

```bash
$ pytest --runxfail xfail_test.py 
================================================================ test session starts =================================================================
platform linux -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/taylermulligan/pytest_xfail
plugins: check-2.4.1
collected 1 item                                                                                                                                     

xfail_test.py F                                                                                                                                [100%]

====================================================================== FAILURES ======================================================================
_____________________________________________________________________ test_xfail _____________________________________________________________________

check = <pytest_check.context_manager.CheckContextManager object at 0x7f2db410d120>

    @pytest.mark.xfail(reason="foo")
    def test_xfail(check):
>       assert False
E       assert False

xfail_test.py:5: AssertionError
============================================================== short test summary info ===============================================================
FAILED xfail_test.py::test_xfail - assert False
================================================================= 1 failed in 0.01s ==================================================================
```

# Actual behaviour

```bash
$ pytest --runxfail xfail_test.py 
================================================================ test session starts =================================================================
platform linux -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/taylermulligan/pytest_xfail
plugins: check-2.4.1
collected 1 item                                                                                                                                     

xfail_test.py x                                                                                                                                [100%]

================================================================= 1 xfailed in 0.03s =================================================================
```

--- a/src/pytest_check/plugin.py
+++ b/src/pytest_check/plugin.py
@@ -20,7 +20,7 @@ def pytest_runtest_makereport(item, call):
     check_log.clear_failures()
 
     if failures:
-        if item._store[xfailed_key]:
+        if item._store[xfailed_key] and not item.config.option.runxfail:
             report.outcome = "skipped"
             report.wasxfail = item._store[xfailed_key].reason
         else:

--- a/tests/test_xfail.py
+++ b/tests/test_xfail.py
@@ -24,3 +24,9 @@ def test_xpass_strict(pytester):
     result = pytester.runpytest("test_example_xfail.py::test_xfail_pass_strict")
     result.assert_outcomes(failed=1)
     result.stdout.fnmatch_lines(["* 1 failed *"])
+
+def test_xfail_runxfail(pytester):
+    pytester.copy_example("examples/test_example_xfail.py")
+    result = pytester.runpytest("--runxfail", "test_example_xfail.py")
+    result.assert_outcomes(passed=2, failed=2)
+    result.stdout.fnmatch_lines(["* 2 failed, 2 passed *"])

